apps-script/contact-form-app/rolodex.gs

```diff
@@ -0,0 +1,341 @@
+                                 {"key": "contactType",
+                                  "value": type
+                    }]}}}]}}]};
+  // Returns another dialog with contact information that the user input.
+  if (fromDialog) {
+    return {
+    "action_response": {
+      "type": "DIALOG",
+      "dialog_action": {
+        "dialog": {
+          "body": {
+            "sections": [ cardConfirmation ]
+          }
+        }
+      }
+    }
+  };
+  }
+  // Updates existing card message with contact information that the user input.
+  else return{
+      "actionResponse": {
+      "type": "UPDATE_MESSAGE",
+      },
+    "privateMessageViewer": user,
+    "cardsV2": [{
+      "card": {"sections": [cardConfirmation]}
+      }]
+  }
+  
+}
+
+/**
+* Submits information from a dialog or card message.
+*
+* @param {Object} user the person who submitted the information.
+* @param {Object} userInputs the form input values specified in event object parameters
+* @param {boolean} whether information was submitted from a dialog
+*
+* @return {Object} 
+*/
+function submitForm(user, userInputs, dialogEventType) {
+  console.log('Submitting form inputs: ' + JSON.stringify(userInputs));
+  const contactName = userInputs["contactName"]
+  
+  // Checks to make sure the user entered a contact
+  // name. If no name value detected, returns
+  // an error message.
+  
+  if (!contactName) {
+    const errorMessage = "Don't forget to name your new contact!"
+    if (dialogEventType == "SUBMIT_DIALOG") {
+      return {
+        "actionResponse": {
+          "type": "DIALOG",
+          "dialogAction": {
+            "actionStatus": {
+              "statusCode": "INVALID_ARGUMENT",
+              "userFacingMessage": errorMessage
+            }
+          }
+        }
+      };
+    } else {
+      return {
+        "privateMessageViewer": user,
+        "text": errorMessage 
+      }
+
+    }
+
+    // Otherwise the Chat app indicates that it received
+    // form data from the dialog or card. Sends private
+    // text message that confirms submission.
+  } else {
+    const confirmationMessage = "✅ " + contactName + " has been added to your contacts."
+    if (dialogEventType == "SUBMIT_DIALOG") {
+      return {
+        "actionResponse": {
+          "type": "NEW_MESSAGE",
+          "dialogAction": {
+            "actionStatus": {
+              "statusCode": "OK",
+              "userFacingMessage": "Success " + JSON.stringify(contactName)
+            }
+          }
+        },
+        "privateMessageViewer": user,
+        "text": confirmationMessage
+      }} else {
+        return {
+        "actionResponse": {
+          "type": "NEW_MESSAGE"},
+        "privateMessageViewer": user,
+        "text": confirmationMessage
+    }}
+  }
+}
+
+/**
+* Converts date in milliseconds since epoch to user-friendly string.
+*
+*
+* @param {Object} milliseconds since epoch time.
+*
+* @return {string} Display-friend date (English US)
+*/
```

Comment:
```suggestion
/**
  * Converts date in milliseconds since epoch to user-friendly string.
  *
  * @param {Object} millis the milliseconds since epoch time.
  * @return {string} Display-friend date (English US)
  */
```